src/virtual_memory.cpp

```diff
@@ -93,8 +93,12 @@ void* allocMemoryPages(std::size_t bytes) {
 	mem = VirtualAlloc(nullptr, bytes, MEM_COMMIT, PAGE_READWRITE);
 	if (mem == nullptr)
 		throw std::runtime_error(getErrorMessage("allocMemoryPages - VirtualAlloc"));
+#else
+#ifdef __APPLE__
+	mem = mmap(nullptr, bytes, PAGE_READWRITE, MAP_ANONYMOUS | MAP_PRIVATE | MAP_JIT, -1, 0);
```

Comment:
It would be a slight bit cleaner to do something like
```
#ifdef __APPLE__
#define MAPFLAGS MAP_ANONYMOUS|MAP_PRIVATE|MAP_JIT
#else 
#define MAPFLAGS MAP_ANONYMOUS|MAP_PRIVATE
#endif 
     mem = mmap(nullptr, bytes, PAGE_READWRITE, MAPFLAGS, -1, 0);
```

Reply:
Another alternative
```
#ifdef __APPLE__
#define MEXTRA MAP_JIT
#else 
#define MEXTRA 0
#endif 
    mem = mmap(nullptr, bytes, PAGE_READWRITE, MAP_ANONYMOUS | MAP_PRIVATE | MEXTRA, -1, 0);

```